api/apps/api/src/modules/scenarios/scenarios-crud.service.ts

```diff
@@ -218,4 +221,29 @@ export class ScenariosCrudService extends AppBaseService<
 
     return query;
   }
+
+  async createWithProjectScenarioId(
+    dto: CreateScenarioDTO,
+    info?: AppInfoDTO,
+  ): Promise<Either<typeof scenarioNotCreated, Scenario>> {
+    const scenario = await this.create(dto, info);
+    try {
+      await this.repository.query(
+        `UPDATE scenarios
+          SET project_scenario_id = (SELECT COALESCE(MAX(sub.project_scenario_id), 0) FROM scenarios sub WHERE sub.project_id = $1) + 1
+          WHERE id = $2
+          RETURNING *`,
+        [scenario.projectId, scenario.id],
+      );
+
+      return right(
+        await this.repository.findOneOrFail({
+          where: { id: scenario.id },
+        }),
+      );
+    } catch (e) {
+      await this.repository.delete(scenario.id);
+      return left(scenarioNotCreated);
+    }
+  }
 }
```

Comment:
something like this should also work _and_ would offload some logic to postgresql, so for example in the case of cloned projects we should not need to handle `project_scenario_numeric_id` in piece exporters and piece importers, as long as we're ok with numeric ids potentially not being consistent between clones of the same project, but I think that's fine because if users delete some scenarios from a cloned projects and then create new ones, in some cases numids may be "recycled" as we don't keep track (nor should, in my view) of already-assigned numids.

When importing a project and its scenarios, numids would be automatically assigned on insert.

So far we only expect these numids to be used to associate best and summed solutions to individual scenarios, so I don't think that having inconsistent numids between clones should be a problem.

```
CREATE OR REPLACE FUNCTION next_numid_for_scenario_in_project()
RETURNS TRIGGER AS $$
DECLARE
  numid integer;
BEGIN
    numid := (
        SELECT COALESCE(MAX(project_scenario_numeric_id), 0) + 1
        FROM scenarios
        WHERE project_id = NEW.project_id
    );
    NEW.project_scenario_numeric_id := numid;
    RAISE NOTICE 'Setting numid of new scenario % to %', NEW.id, numid;
    RETURN NEW;
END;
$$ LANGUAGE plpgsql;

CREATE TRIGGER set_numid_for_scenario_in_project
BEFORE INSERT ON scenarios
FOR EACH ROW
EXECUTE FUNCTION next_numid_for_scenario_in_project();
```
If going trigger-happy, I think `BEFORE INSERT` should be enough - combined with a non null constraint and unique constraint, there should never be a need to update a scenario's numid when a scenario is updated (e.g. its name is changed)